### ISSUE TITLE
[Bug] dependencies attribute doesn't work inside repeatable field

### ISSUE THREAD

**Issue by DanielKimmich** [Laravel-Backpack/CRUD#2865]
# Bug report
With the repeatable field I have the following problems:

The "id" attribute only creates it in the first repeated field and the other fields not because it is duplicated.

The 'dependencies' attribute field of type relationship does not work on repeatable.

### What I did

        $this->crud->addField([
            'name'  => 'relation_address',
            'label' => trans('contact.address.titles'),
            'type'  => 'repeatable',
            'tab'   => trans('contact.address.tab'),
            'fields' => [

                [   'name'  => 'data8',
                    'label' => trans('contact.address.division'),
                    'type'  => 'relationship',
                    'wrapper' => ['class' => 'form-group col-md-6'],
                    'attributes' => ['id' => 'address_division'],
                    'model' => 'App\Models\WorldDivision', // foreign key model
                    'entity'    => 'addresses',
                    'attribute' => 'name',
                    'ajax' => true,
                    'multiple' => false,
                    'data_source'  => backpack_url('contactperson/fetch/division'), // url to controller
                    'dependencies'  => ['data10'], //this select2 is reset to null
                    'minimum_input_length' => 0, // minimum before querying results
                    'default' => Config::get('settings.contact_division'),
                     ],

### What I expected to happen

Each data8 field has its own id

### What happened

The problem I am having is that I need to filter one select (division) depending on another (country). What exists is deletes the select child (division) if the parent (country) is modified using 'dependencies', it would be nice to have a filter property inside the select filter.

![image](https://user-images.githubusercontent.com/44840413/82665288-da571400-9c09-11ea-9559-51e3d7118761.png)

### What I've already tried to fix it

For now I'm doing it like this, but it works without using a repeatable field.

    public function fetchDivision()
    {
         return $this->fetch([
            'model' => 'App\Models\WorldDivision', // required
            'searchable_attributes' => ['name'],
            'paginate' => 50, // items to show per page
            'query' => function($model) {
                $form = $this->crud->getRequest()->input('form');
                foreach ($form as $entry) {
                    if ($entry['name'] == 'country_id') {
                        $country_id = (int) $entry['value'];
                        break 1;  // Sólo sale del foreach. 
                    }
                }
                return $model->where('country_id', $country_id)->orderBy('name');
            } // to filter the results that are returned
        ]); 
    }

### PHP VERSION:
PHP 7.2.11 (cli) (built: Oct 10 2018 02:04:07) ( ZTS MSVC15 (Visual C++ 2017) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

### LARAVEL VERSION:
v7.12.0@c2fff1e9879494a6f853593b3c517dc9922bbb51

### BACKPACK VERSION:
4.1.5@e2c166b065f6b405747db5a5f4d36a2a88f8dc22

I raise this need as there may be a more comprehensive solution and possibly not as complex as I want to do using the "id" attribute

**Comment by tabacitu** [comment]
Hi @DanielKimmich ,

Thanks for taking the time to submit this. We'll investigate dependencies inside `relationship` inside `repeatable` and come back ASAP. Until then, about this:

> The "id" attribute only creates it in the first repeated field and the other fields not because it is duplicated.

If I understand correctly, this is to be expected... right? I mean, that's the point of fields inside the repeatable, to get cloned/copied. So indeed, I guess you can't use the id attribute on it, that will also get cloned. I don't see a way to "fix" this to be honest. What are you trying to achieve with id on a field inside repeatable?

Cheers!

**Comment by DanielKimmich** [comment]
Thank you for viewing this case. What I want to do is to be able to obtain the value of the select "pais" (country) to be able to filter it in "provincia" (division). See in the previous image that passes. And I don't know how to get the value of your pair. So I wanted to use an "id" in the field. I think there must be another easier way to do it, but I can't think of it.

**Comment by pxpm** [comment]
Hey @DanielKimmich \o

In an ajax field you can submit form inputs along with the search process using `include_all_form_fields => true`.

Dependencies does not depend on id but rather in name, anyway they are both stripped in repeatable. So it's a no go for dependencies in repeatables at the moment.

We already add a custom attribute with the name we stripe from the field in repeatable: `data-repeatable-input-name`.

So maybe we can change the dependencies function to work with both of them.

Currently we have:

```php
            $('input[name='+$dependency+'], select[name='+$dependency+'], checkbox[name='+$dependency+'], radio[name='+$dependency+'], textarea[name='+$dependency+']').change(function () {
                element.val(null).trigger("change");
            });

```
Something along this lines could work in repeatable:

```php
            $('input[name='+$dependency+'], input[data-repeatable-input-name='+$dependency+'], select[name='+$dependency+'],select[data-repeatable-input-name='+$dependency+'] ...
```
But this has a **major drawback** !! In the repeatable could exist more than one `data-repeatable-input-name` with same value, so when clearing one it would clear them all.

Maybe changing the way we pick the inputs to use `closest()` might work. But atm I don't have a solution for this problem.

Let me know if you find a working solution for it.

For the FetchOperation question you could use something like: `$form = collect($request->input('form'))->pluck('value', 'name');`
so you can avoid the foreach loop and just use: `isset($form['field']) && !is_null($form['field']` ...

Best,
Pedro

**Comment by pxpm** [comment]
closed in favor of #3218
